cpu/stm32/periph/flashpage.c

```diff
@@ -118,8 +126,19 @@ static void _erase_page(void *page_addr)
     pn = (uint8_t)page;
 #endif
     CNTRL_REG &= ~FLASH_CR_PNB;
+#if defined(CPU_FAM_STM32F4) || defined(CPU_FAM_STM32F7)
+    if (FLASHPAGE_DUAL_BANK && (pn > (FLASHPAGE_NUMOF / 2 - 1))) {
+        pn = pn - (FLASHPAGE_NUMOF / 2);
+        CNTRL_REG |= FLASH_CR_SNB_4 | (uint32_t)(pn << FLASH_CR_PNB_Pos);
+    }
+    else {
+        CNTRL_REG |= (uint32_t)(pn << FLASH_CR_PNB_Pos);
+    }
+#else
     CNTRL_REG |= (uint32_t)(pn << FLASH_CR_PNB_Pos);
+#endif
```

Comment:
```suggestion
```

We can get rid of that `#ifdef`, makes it less fragile when adding new STM32 siblings.

Reply:
See above